streaming/base/sampling.py

```diff
@@ -23,6 +23,18 @@ def get_sampling(samples_per_shard: NDArray[np.int64], choose: int, granularity:
     Returns:
         NDArray[np.int64]: Array of ephemeral samples chosen per shard.
     """
+    if choose < 0:
+        raise ValueError('Choose must be a non-negative integer.')
+
+    if granularity <= 0:
+        raise ValueError('Granularity must be a positive integer.')
+
+    if seed < 0:
```

Comment:
Same here. Have it in dataset.py

Reply:
done

future work: check the required bounds of python random, numpy, torch, etc. "set seed" methods. would be a barrel of monkeys if they are different. then tighten bounds as appropriate. wondering if it's 0 <= seed < (1 << 31)